pkg/mcclient/mcclient.go

```diff
@@ -21,46 +21,47 @@ package mcclient
 import (
 	"context"
 	"errors"
-	"reflect"
+	"fmt"
 	"sync"
+	"time"
 
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/fields"
-	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/client-go/rest"
 	"k8s.io/client-go/tools/cache"
 	"k8s.io/client-go/tools/clientcmd"
-	log "k8s.io/klog/v2"
 
 	"github.com/kcp-dev/edge-mc/pkg/apis/edge/v1alpha1"
 	ksclientset "github.com/kcp-dev/edge-mc/pkg/client/clientset/versioned"
 	ksinformers "github.com/kcp-dev/edge-mc/pkg/client/informers/externalversions"
 	mcclientset "github.com/kcp-dev/edge-mc/pkg/mcclient/clientset"
-	"github.com/kcp-dev/edge-mc/pkg/mcclient/listwatch"
 )
 
 type KubestellarClusterInterface interface {
 	// Cluster returns clientset for given cluster.
 	Cluster(name string) (client mcclientset.Interface)
 	// ConfigForCluster returns rest config for given cluster.
 	ConfigForCluster(name string) (*rest.Config, error)
-	// CrossClusterListWatch returns cross-cluster ListWatch
-	CrossClusterListWatch(gv schema.GroupVersion, resource string, namespace string, fieldSelector fields.Selector) *listwatch.CrossClusterListerWatcher
 }
 
 type multiClusterClient struct {
-	configs    map[string]*rest.Config
-	clientsets map[string]*mcclientset.Clientset
-	lock       sync.Mutex
+	ctx              context.Context
+	configs          map[string]*rest.Config
```

Comment:
Can't we get the config from the clientset in case someone needs it?

Reply:
I don't see an easy way to do this.